Sample program PolyEditor.py aborts

**Operating system**: OS X El Capitan 10.11.6
**wxPython version & source**: 4.0.7.post2 (pip)
**Python version & source**: 3.6.2 (v3.6.2:5fd33b5926, Jul 16 2017, 20:11:06) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]

**Description of the problem**:

Source:

https://github.com/wxWidgets/Phoenix/blob/master/samples/floatcanvas/PolyEditor.py

What happens:

The program aborts before showing the GUI.

Output:

18:19:54: Debug: Adding duplicate image handler for 'PNG file'
18:19:54: Debug: Adding duplicate image handler for 'JPEG file'
18:19:54: Debug: Adding duplicate image handler for 'TIFF file'
18:19:54: Debug: Adding duplicate image handler for 'GIF file'
18:19:54: Debug: Adding duplicate image handler for 'PNM file'
18:19:54: Debug: Adding duplicate image handler for 'PCX file'
18:19:54: Debug: Adding duplicate image handler for 'IFF file'
18:19:54: Debug: Adding duplicate image handler for 'Windows icon file'
18:19:54: Debug: Adding duplicate image handler for 'Windows cursor file'
18:19:54: Debug: Adding duplicate image handler for 'Windows animated cursor file'
18:19:54: Debug: Adding duplicate image handler for 'TGA file'
18:19:54: Debug: Adding duplicate image handler for 'XPM file'
18:19:54: Debug: wxColour::Set - couldn't set to colour string 'LightBlue'
Traceback (most recent call last):
  File "PolyEditor.py", line 224, in OnInit
    frame.Setup()
  File "PolyEditor.py", line 165, in Setup
    self.Canvas.ZoomToBB()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/wx/lib/floatcanvas/FloatCanvas.py", line 802, in ZoomToBB
    self.SetToNewScale(DrawFlag=DrawFlag)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/wx/lib/floatcanvas/FloatCanvas.py", line 821, in SetToNewScale
    self.Draw()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/wx/lib/floatcanvas/FloatCanvas.py", line 628, in Draw
    self._DrawObjects(dc, self._DrawList, ScreenDC, self.ViewPortBB, HTdc)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/wx/lib/floatcanvas/FloatCanvas.py", line 986, in _DrawObjects
    Object._Draw(dc, WorldToPixel, ScaleWorldToPixel, HTdc)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/wx/lib/floatcanvas/FCObjects.py", line 697, in _Draw
    dc.SetBrush(self.Brush)
wx._core.wxAssertionError: C++ assertion "retval != __null" failed at /Users/robind/projects/bb2/dist-osx-py36/build/ext/wxWidgets/src/osx/carbon/graphics.cpp(141) in wxMacCreateCGColor(): 
OnInit returned false, exiting...

--- a/samples/floatcanvas/MovingPlot.py
+++ b/samples/floatcanvas/MovingPlot.py
@@ -220,7 +220,6 @@ class DemoApp(wx.App):
     """
 
     def OnInit(self):
-        wx.InitAllImageHandlers()
         frame = DrawFrame(None, wx.ID_ANY,
                           title = "Plotting Test",
                           size = (700,400) )

--- a/samples/floatcanvas/PolyEditor.py
+++ b/samples/floatcanvas/PolyEditor.py
@@ -153,7 +153,7 @@ class DrawFrame(wx.Frame):
         Range = (-10,10)
 
         # Create a couple of random Polygons
-        for i, color in enumerate(("LightBlue", "Green", "Purple","Yellow")):
+        for i, color in enumerate(("Light Blue", "Green", "Purple","Yellow")):
             points = RandomArray.uniform(Range[0],Range[1],(6,2))
             Poly = self.Canvas.AddPolygon(points,
                                      LineWidth = 2,

--- a/samples/floatcanvas/PolyEditor.py
+++ b/samples/floatcanvas/PolyEditor.py
@@ -210,7 +210,6 @@ class PolyEditor(wx.App):
     """
 
     def OnInit(self):
-        wx.InitAllImageHandlers()
         frame = DrawFrame(None,
                           -1,
                           "FloatCanvas Demo App",
@@ -224,7 +223,7 @@ class PolyEditor(wx.App):
         frame.Setup()
         return True
 
-PolyEditor(0).MainLoop()# put in True if you want output to go to it's own window.
+PolyEditor().MainLoop()
 
 
 

